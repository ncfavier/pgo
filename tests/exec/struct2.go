package main

import "fmt"

func main() {
	z := new(L)
	z.x = 42
	fmt.Print(z.x)
	fmt.Print(z.next)
	fmt.Print("\n")
	z.next = new(L)
	n := z.next
	n.x = 43;
	fmt.Print(z.x)
	fmt.Print(z.next.x)
	fmt.Print(z.next.next)
	fmt.Print("\n")
}

type L struct {
	x    int
	next *L
}

