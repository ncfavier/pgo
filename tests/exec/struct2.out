42<nil>
4243<nil>
